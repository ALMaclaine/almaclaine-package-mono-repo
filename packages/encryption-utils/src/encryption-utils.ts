import {hash, compare} from 'bcrypt';

export async function encryptPassword(password: string, rounds: number = 64000) {
    return hash(password, rounds);
}

export async function comparePassword(providedPassword: string, actualPassword: string) {
    return compare(providedPassword, actualPassword);
}
