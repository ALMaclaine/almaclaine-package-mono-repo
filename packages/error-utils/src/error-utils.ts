export class ALMError extends Error {
    constructor(message: string, public readonly type: string) {
        super(message);
    }
}
