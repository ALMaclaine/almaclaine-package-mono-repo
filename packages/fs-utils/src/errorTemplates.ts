export const doesNotExistError = (path: string): string => `Provided path ${path} does not exist: ${path}.`;

export const alreadyExistsError = (path: string): string => `Provided file/directory already exists: ${path}.`;

export const isNotDirectoryError = (path: string): string => `Provided path ${path} is not a directory.`;

export const isDirectoryError = (path: string): string => `Provided path ${path} is a directory.`;

export const isNotJsonFilError = (path: string): string => `Provided path ${path} is not a json file. Extension must be .json`;

