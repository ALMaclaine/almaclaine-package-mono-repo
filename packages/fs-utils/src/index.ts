export * from './fs-utils';
export * from './errorTemplates';
