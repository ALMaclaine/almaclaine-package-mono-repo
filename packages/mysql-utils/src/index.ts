export * from './mysql-utils';
export type {ConnectionInfo} from './types';
