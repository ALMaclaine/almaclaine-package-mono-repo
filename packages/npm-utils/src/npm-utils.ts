import {
    alreadyExistsError,
    existsSync,
    join,
    makeCwdRelPath,
    writeFileAsStringSync,
} from '@almaclaine/fs-utils';

import validPackage from "validate-npm-package-name";

import {unlinkSync} from "fs";

import readline from "readline-promise";
import {execSync} from "child_process";

const execSyncToString = (cmd: string) => execSync(cmd).toString('utf-8').trim();

export async function initPackage(folderName: string) {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    if (!validPackage(folderName)) {
        throw Error('Package name must be in the form of word(-word)*.');
    }
    const dirPath = makeCwdRelPath(folderName);

    execSyncToString(`npx degit almaclaine/typescript-package-template ${dirPath}`);

    if (!existsSync(dirPath)) {
        throw Error(`Failed to create directory: ${folderName}.\n${alreadyExistsError(dirPath)}.`);
    }

    unlinkSync(join(dirPath, 'src', 'typescript-package-template.ts'));
    writeFileAsStringSync(join(dirPath, 'src', `${folderName}.ts`), '');

    unlinkSync(join(dirPath, 'tests', 'typescript-package-template.test.ts'));
    writeFileAsStringSync(join(dirPath, 'tests', `${folderName}.test.ts`), '');

    const desc = await rl.questionAsync('Enter a description for the package: ');
    const author = await rl.questionAsync('Package Author: ');
    const keywords = await rl.questionAsync('Enter package keywords separated by space: ');
    const git = await rl.questionAsync('Github url (no .git, optional): ');

    const pack = require(join(dirPath, 'package.json'));
    pack.name = `${folderName}`;
    pack.description = desc;
    pack.author = `${author.trim()}`;
    if (git) {
        pack.repository.url = git + '.git';
        pack.bugs.url = git + '/issues';
        pack.homepage = git;
        execSync(`cd ${dirPath}; git init; git remote add origin ${git}.git`);
    }

    // Capitalize first letter of each Keyword, Don't Allow Duplicates
    pack.keywords = [...new Set(keywords.trim()
        .replace(/\s\s+/g, ' ')
        .split(' ')
        .map(e => e.toLowerCase())
        .map(e => e[0].toUpperCase() + e.slice(1)))];
    writeFileAsStringSync(join(dirPath, 'package.json'), JSON.stringify(pack, null, 2));
    rl.close();
}
