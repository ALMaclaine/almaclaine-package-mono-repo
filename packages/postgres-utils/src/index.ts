export * from './postgres-utils';
export * from './ConnectionManager';
