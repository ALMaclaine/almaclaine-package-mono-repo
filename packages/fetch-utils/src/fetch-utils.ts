const fetch = require('cross-fetch');
import {ALMError} from "@almaclaine/error-utils";

export const ErrorTypes = {
    FetchFailed: 'FetchFailed'
}

export class FetchFailedError extends ALMError {
    constructor(message: string) {
        super(message, ErrorTypes.FetchFailed);
    }
}

export async function almFetch(url: string, options: {[key: string]: any} = {}) {
    const res = await fetch(url, options);
    if(!res.ok) {
        throw new FetchFailedError(`Fetch failed with status code: ${res.status}`);
    }
    return res;
}
