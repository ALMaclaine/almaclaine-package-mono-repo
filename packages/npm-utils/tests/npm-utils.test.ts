import * as npmUtils from '../src';

describe('utils/npm-utils', () => {
    it('Has Correct API', () => {
        const keys = Object.keys(npmUtils);
        expect(keys.length).toBe(1);
        expect(keys).toMatchSnapshot();
    });
});
