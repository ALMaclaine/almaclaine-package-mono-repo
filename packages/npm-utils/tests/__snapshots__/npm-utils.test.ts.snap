// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/npm-utils Has Correct API 1`] = `
Array [
  "initPackage",
]
`;
